src/charm.py

```diff
@@ -421,6 +471,11 @@ def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
         logger.info("Pebble plan updated with new configuration, replanning")
         self._container.replan()
 
+        if not self._log_level_valid:
+            logger.info(f"Invalid configuration value for log_level: {self._log_level}")
+            self.unit.status = BlockedStatus("Bad configuration value for log_level")
+            return
```

Comment:
This could be a function to avoid code duplication (`_validate_config`?). Shouldn't this event (and some of the other events) be deferred if the configuration is bad?